### PR TITLE
Refactor and extend GPU metrics error types

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Refactor and extend GPU metrics error types to better track missing GPUs.

### CODE REVIEW

**Comment by aleksandra-malinowska** [comment]
cc @kgolab  - a couple of last-minute changes, let me know if I'm breaking something :)

File: cluster-autoscaler/utils/gpu/gpu.go [hunk 1]
```diff
@@ -95,45 +96,44 @@ func FilterOutNodesWithUnreadyGpus(allNodes, readyNodes []*apiv1.Node) ([]*apiv1
...
 	gpuType, labelFound := node.Labels[GPULabel]
 	capacity, capacityFound := node.Status.Capacity[ResourceNvidiaGPU]
 
+	if !labelFound {
+		// no label, fallback to generic solution
+		if capacityFound && !capacity.IsZero() {
+			return MetricsGenericGPU
+		}
+
+		// no signs of GPu
```
Comment (kgolab) [inline cluster-autoscaler/utils/gpu/gpu.go]: /nit

**Comment by MaciekPytel** [comment]
At this point (multiple different error states), I'd define a separate 'state' label, rather than push all the different errors as fake 'gpu types'. That being said this is not doing anything the previous version wasn't doing already and it's not worth the effort to refactor this whole thing again.

/lgtm

**Comment by kgolab** [comment]
LGTM, too.
As discussed off-line I understand that we don't want to merge "missing GPU" with label value to have both error & name of the problematic GPU.
Maybe introducing separate state would indeed be good but let's keep it simple for now.

**Pull request merged**